tuna/miopen/subcmd/export_db.py

```diff
@@ -428,25 +508,65 @@ def export_pdb(dbt: MIOpenDBTables, args: argparse.Namespace,
   return local_path
 
 
-def populate_sqlite(cfg_map, num_perf, cnx, perf_db_entry, cfg_entry,
-                    total_entries, logger: logging.Logger):
-  """Analyze perf_dv entry"""
-  if cfg_entry.id in cfg_map:
-    ins_cfg_id = cfg_map[cfg_entry.id]
-  else:
-    cfg_dict = get_sqlite_cfg_dict(perf_db_entry.fdb_key)
+def build_miopen_pdb(query, logger: logging.Logger) -> OrderedDict:
+  """return dict with key: fdb_key, val: list of fdb entries"""
+  perf_db: OrderedDict = OrderedDict()
+  solvers: Dict[str, Dict[str, Any]] = {}
+  db_key_map: Dict[str, str] = {}
+  db_entries = query.all()
+  total_entries = len(db_entries)
+  logger.info("pdb query returned: %s", total_entries)
 
-    #filters cfg_dict by SQLITE_CONFIG_COLS, inserts cfg if missing
-    ins_cfg_id = get_config_sqlite(cnx, cfg_dict)
-    cfg_map[cfg_entry.id] = ins_cfg_id
+  cfg_lst = []
+  for _, config in db_entries:
+    if config not in cfg_lst:
+      cfg_lst.append(config)
 
-  pdb_dict = insert_perf_db_sqlite(cnx, perf_db_entry, ins_cfg_id)
-  num_perf += 1
+  db_key_map = fin_db_key(cfg_lst, logger)
 
-  if num_perf % (total_entries // 10) == 0:
-    cnx.commit()
-    logger.info("PDB count: %s, mysql cfg: %s, pdb: %s", num_perf, cfg_entry.id,
-                pdb_dict)
+  for pdb_entry, config in db_entries:
+    if add_entry_to_solvers(pdb_entry, solvers, logger):
+      db_key = db_key_map[config.id]
+      lst = perf_db.get(db_key)
+      if not lst:
+        perf_db[db_key] = [pdb_entry]
+      else:
+        lst.append(pdb_entry)
+
+  return perf_db
+
+
+def write_pdb(arch, num_cu, ocl, perf_db, filename=None):
```

Comment:
probably not a good idea to default filename to None it seems like it needs to be not none based on your code